containers/signing-service/Dockerfile.j2

```diff
@@ -3,19 +3,19 @@ FROM debian:{{ debian_version }}
 COPY create-user.sh                     /tmp/
 RUN  /tmp/create-user.sh
 
-COPY signing-service/RELEASE-VERSION    /usr/lib/signing_service/RELEASE-VERSION
-COPY signing-service/requirements.lock  /usr/lib/signing_service/signing-service/
-COPY testing.pref                       /etc/apt/preferences.d/
-COPY build-dependencies.txt             /tmp/
-COPY install-dependencies.sh            /tmp/
-COPY install-signing-service.sh         /tmp/
-COPY clean-container.sh                 /tmp/
-COPY middleman_protocol/                /usr/lib/signing_service/middleman_protocol/
-COPY signing-service/RELEASE-VERSION    /usr/lib/signing_service/middleman_protocol/RELEASE-VERSION
+COPY signing-service/cacheable-requirements.lock  /usr/lib/signing_service/signing-service/
+COPY testing.pref                                 /etc/apt/preferences.d/
+COPY build-dependencies.txt                       /tmp/
+COPY install-dependencies.sh                      /tmp/
+COPY install-signing-service.sh                   /tmp/
+COPY clean-container.sh                           /tmp/
 RUN  /tmp/install-signing-service.sh
+COPY middleman_protocol                           /usr/lib/signing_service/middleman_protocol/
+COPY install-middleman-protocol.sh                /tmp/
+RUN  /tmp/install-middleman-protocol.sh
```

Comment:
`middleman_protocol` should be before Signing Service. It's going to change much less often.

Reply:
This optimization separate `middleman_protocol` because it's always change. Why do you think the `singing service` is changing more than `middleman_protocol` ?

Reply:
Because it depends on it.

But I see that `install-signing-service.sh` mostly installs its dependencies. In that case it makes sense for it to be before but perhaps we should rename the script it make it clear.